pkg/cache/v2/snapshot.go

```diff
@@ -31,6 +31,21 @@ type Resources struct {
 	Items map[string]types.ResourceWithTtl
 }
 
+// DeltaResources is a versioned group of resources which also contains individual resource versions per the incremental xDS protocol
+type DeltaResources struct {
+	// Version information
+	SystemVersion string
+
+	// Items in the group indexed by name
+	Items resourceItems
+}
+
+// resourceItems contain the lower level versioned resource map
+type resourceItems struct {
+	Version string
+	Items   map[string]types.Resource
```

Comment:
Maybe use ResourceWithTTL here? I don't think we need to support TTL with Delta in this PR, but might as well use the right types if its not too much work

Reply:
👍 I'll move to the `ResourceWithTTL` here. When I wrote this code that wasn't in yet and I never updated so I'll change it now while we're working on it